File: files/CH-073.R
library(tidyverse)
library(readxl)

path = "files/CH-073 Custom splitter 2.xlsx"
input = read_xlsx(path, range = "B2:B15")
test  = read_xlsx(path, range = "D2:F24")

date_pattern = "[0-9]{4}\\/[0-9]{1,2}\\/[0-9]{1,2}"
product_quant_pattern = "([A-Z]+[0-9]+)"

result = input %>%
  mutate(
    Date = str_extract(Info, date_pattern),
    Info2 = str_remove(Info, date_pattern),
    prod_quant = map(Info2, ~ unlist(str_extract_all(.x, product_quant_pattern)))
  ) %>%
  unnest(prod_quant) %>%
  select(Date, prod_quant) %>%
  extract(prod_quant, into = c("Product", "Quantity"), regex = "([A-Z]+)([0-9]+)") %>%
  mutate(Quantity = as.numeric(Quantity))

identical(result, test)
# [1] TRUE

File: files/000-099/073/CH-073.py
import pandas as pd
import re

path = "CH-073 Custom Splitter 2.xlsx"
input = pd.read_excel(path, usecols= "B", skiprows=1, nrows = 13)
test  = pd.read_excel(path, usecols= "D:F", skiprows=1)

date_pattern = r"\d{4}/\d{1,2}/\d{1,2}"
product_quant_pattern = r"([A-Z]+\d+)"

input['Date'] = input['Info'].apply(lambda x: re.search(date_pattern, x).group())
input['Info2'] = input['Info'].apply(lambda x: re.sub(date_pattern, '', x))
input['prod_quant'] = input['Info2'].apply(lambda x: re.findall(product_quant_pattern, x))
input = input.explode('prod_quant')
input[['Product', 'Quantity']] = input['prod_quant'].str.extract(r"([A-Z]+)(\d+)")
input['Quantity'] = input['Quantity'].astype("int64")
result = input[['Date', 'Product', 'Quantity']].reset_index(drop=True)

print(result.equals(test)) # True